File: Windows8MVVM_FinalSourceCode/Chapter6/FinanceHub_ViewModelViewModel/FinanceHub/FinanceHub/Model/Stock.js
(function () {
    "use strict";

    WinJS.Namespace.define("FinanceHub.Model", {
        Stock: WinJS.Class.define(
            function (symbol, currentPrice, openPrice, change, daysRange, range52Week) {

                this.Symbol = symbol;
                this.OpenPrice = openPrice;
                this.CurrentPrice = currentPrice;
                this.Change = change;
                this.DaysRange = daysRange;
                this.Range52Week = range52Week;
            }
            , {

                Symbol: undefined,
                OpenPrice: undefined,
                CurrentPrice: undefined,
                Change: undefined,
                DaysRange: undefined,
                Range52Week: undefined
            }),
    });
})();

File: Windows8MVVM_FinalSourceCode/Chapter3/FinanceHub/FinanceHub/Model/Stock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using System.Runtime.Serialization;

namespace FinanceHub.Model
{
    [KnownType(typeof(FinanceHub.Model.Stock))]
    [DataContractAttribute]
    public class Stock
    {
        [DataMember()]
        public string Symbol { get; set; }
        [DataMember()]
        public decimal CurrentPrice { get; set; }
        [DataMember()]
        public decimal OpenPrice { get; set; }
        [DataMember()]
        public double Change { get; set; }
        [DataMember()]
        public string DaysRange { get; set; }
        [DataMember()]
        public string Range52Week { get; set; }
    }
}
